common/utils/utils.go

```diff
@@ -588,91 +588,68 @@ func ReadSpannerSchema(ctx context.Context, conv *internal.Conv, client *sp.Clie
 }
 
 // CompareSchema compares the spanner schema of two conv objects and returns specific error if they don't match
-func CompareSchema(conv1, conv2 *internal.Conv) error {
-	if conv1.SpDialect != conv2.SpDialect {
-		return fmt.Errorf("spanner dialect don't match")
+func CompareSchema(sessionFileConv, actualSpannerConv *internal.Conv) error {
+	if sessionFileConv.SpDialect != actualSpannerConv.SpDialect {
+		return fmt.Errorf("spanner dialect don't match: session dialect %v, spanner dialect %v", sessionFileConv.SpDialect, actualSpannerConv.SpDialect)
 	}
-	for _, sessionTable := range conv1.SpSchema {
-		spannerTableId, _ := internal.GetTableIdFromSpName(conv2.SpSchema, sessionTable.Name)
-		spannerTable := conv2.SpSchema[spannerTableId]
+	for _, sessionTable := range sessionFileConv.SpSchema {
+		spannerTableId, err := internal.GetTableIdFromSpName(actualSpannerConv.SpSchema, sessionTable.Name)
+		if err != nil {
+			return fmt.Errorf("table %v not found in the spanner database schema but found in the session file. If this table does not need to be migrated, please exclude it during the schema conversion process and migration process", sessionTable.Name)
+		}
+		spannerTable := actualSpannerConv.SpSchema[spannerTableId]
+		sessionTableParentName := sessionFileConv.SpSchema[sessionTable.ParentId].Name
+		spannerTableParentName := actualSpannerConv.SpSchema[spannerTable.ParentId].Name
 
-		sessionTableParentName := conv1.SpSchema[sessionTable.ParentId].Name
-		spannerTableParentName := conv2.SpSchema[spannerTable.ParentId].Name
+		//table names should match
+		if sessionTable.Name != spannerTable.Name {
+			return fmt.Errorf("table name don't match: session table %v, spanner table %v", sessionTable.Name, spannerTable.Name)
+		}
+
+		//parent table names should match
+		if sessionTableParentName != spannerTableParentName {
+			return fmt.Errorf("parent table name don't match: session table %v, parent session table name: %v, spanner table %v, parent spanner table name: %v", sessionTable.Name, sessionTableParentName, spannerTable.Name, spannerTableParentName)
+		}
 
-		if sessionTable.Name != spannerTable.Name || sessionTableParentName != spannerTableParentName ||
-			len(sessionTable.PrimaryKeys) != len(spannerTable.PrimaryKeys) || len(sessionTable.ColDefs) != len(spannerTable.ColDefs) ||
```

Comment:
Shouldn't we keep the check len(sessionTable.ColDefs) != len(spannerTable.ColDefs)? What if an extra column is added in spanner ?

Reply:
Right, I must have missed this during refactoring. Thanks adding it back